kroxylicious-kms-providers/kroxylicious-kms-provider-azure-key-vault-kms-test-support/src/main/java/io/kroxylicious/kms/provider/azure/kms/AzureKeyVaultKmsTestKmsFacade.java

```diff
@@ -0,0 +1,233 @@
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.images.PullPolicy;
+import org.testcontainers.utility.DockerImageName;
+
+import com.azure.core.credential.BasicAuthenticationCredential;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpClient;
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.security.keyvault.keys.KeyClient;
+import com.azure.security.keyvault.keys.KeyClientBuilder;
+import com.azure.security.keyvault.keys.models.CreateKeyOptions;
+import com.azure.security.keyvault.keys.models.DeletedKey;
+import com.azure.security.keyvault.keys.models.KeyOperation;
+import com.azure.security.keyvault.keys.models.KeyType;
+import com.azure.security.keyvault.keys.models.KeyVaultKey;
+import com.github.nagyesta.lowkeyvault.http.ApacheHttpClient;
+import com.github.nagyesta.lowkeyvault.http.AuthorityOverrideFunction;
+import com.github.nagyesta.lowkeyvault.http.management.LowkeyVaultException;
+import com.github.nagyesta.lowkeyvault.testcontainers.LowkeyVaultContainer;
+import com.sun.net.httpserver.HttpServer;
+
+import io.kroxylicious.kms.provider.azure.AzureKeyVaultEdek;
+import io.kroxylicious.kms.provider.azure.AzureKeyVaultKmsService;
+import io.kroxylicious.kms.provider.azure.WrappingKey;
+import io.kroxylicious.kms.provider.azure.config.AzureKeyVaultConfig;
+import io.kroxylicious.kms.provider.azure.config.auth.EntraIdentityConfig;
+import io.kroxylicious.kms.service.TestKekManager;
+import io.kroxylicious.kms.service.TestKmsFacade;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.proxy.config.secret.InlinePassword;
+import io.kroxylicious.proxy.config.tls.InsecureTls;
+import io.kroxylicious.proxy.config.tls.Tls;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static com.github.nagyesta.lowkeyvault.testcontainers.LowkeyVaultContainerBuilder.lowkeyVault;
+
+public class AzureKeyVaultKmsTestKmsFacade implements TestKmsFacade<AzureKeyVaultConfig, WrappingKey, AzureKeyVaultEdek> {
+
+    public static final Tls INSECURE_TLS = new Tls(null, new InsecureTls(true), null, null);
+    public static final String MOCK_AUTH_RESPONSE = """
+            {
+                "access_token": "aaa",
+                "token_type": "Bearer",
+                "expires_in": 3599,
+                "scope": "https%3A%2F%2Fgraph.microsoft.com%2Fmail.read",
+                "refresh_token": "bbb",
+                "id_token": "ccc"
+            }
+            """;
+    @Nullable
+    private LowkeyVaultContainer kms;
+    @Nullable
+    private HttpServer entraMock;
+
+    protected AzureKeyVaultKmsTestKmsFacade() {
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return DockerClientFactory.instance().isDockerAvailable();
+    }
+
+    public void startKms() {
+        this.kms = startVault();
+        try {
+            this.entraMock = HttpServer.create();
+            entraMock.createContext("/", exchange -> {
```

Comment:
Can we use WireMockServer here instead or rolling our own?  We've used WireMockServer extensively already and I think it would be better not to diverge.

Reply:
it's all wiremock now